docs/csharp/roslyn-sdk/get-started/syntax-transformation.md

```diff
@@ -0,0 +1,177 @@
+
+You've created the name. Now, it's time to build more nodes into the tree by building a <xref:Microsoft.CodeAnalysis.CSharp.Syntax.QualifiedNameSyntax>. The new tree uses `name` as the left of the name, and a new <xref:Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax> for the `Collections` namespace as the right side of the <xref:Microsoft.CodeAnalysis.CSharp.Syntax.QualifiedNameSyntax>. Add the following code to `program.cs`:
+
+[!code-csharp[create the collections identifier](../../../../samples/csharp/roslyn-sdk/SyntaxTransformationQuickStart/ConstructionCS/Program.cs#CreateQualifiedIdentifierName "Build the System.Collections identifier")]
+
+Run the code again, and see the results. You're building a tree of nodes that represents code. You continue this pattern to build the <xref:Microsoft.CodeAnalysis.CSharp.Syntax.QualifiedNameSyntax> for the namespace `System.Collections.Generic`. Add the following code to `Program.cs`:
+
+[!code-csharp[create the full identifier](../../../../samples/csharp/roslyn-sdk/SyntaxTransformationQuickStart/ConstructionCS/Program.cs#CreateFullNamespace "Build the System.Collections.Generic identifier")]
+
+Run the program again to see that you've build the tree for the code to add.
+
+### Create a modified tree
+
+You've built a small syntax tree that contains one statement. The APIs to create new nodes are the right choice to create single statements, or other small code blocks. However, to build larger blocks of code, you should use methods that replace nodes, or insert nodes into an existing tree. Remember that syntax trees are immutable. The **Syntax API** doesn't provide any mechanism for modifying an existing syntax tree after construction. Instead, the **Syntax API** provides methods that produce new trees based on changes to existing ones. Concrete classes that derive from <xref:Microsoft.CodeAnalysis.SyntaxNode> defines `With*` methods that specify changes to its child properties. Additionally, the <xref:Microsoft.CodeAnalysis.SyntaxNodeExtensions.ReplaceNode%2A> extension method can be used to replace a descendent node in a subtree. This method also updates the parent to point to the newly created child and repeats this process up the entire tree - a process known as _re-spining_ the tree.
+
+The next step is to create a tree that represents an entire (small) program and then modify it. Add the following code to the beginning of the `Program` class:
+
+[!code-csharp[create a parse tree](../../../../samples/csharp/roslyn-sdk/SyntaxTransformationQuickStart/ConstructionCS/Program.cs#DeclareSampleCode "Create a tree that represents a small program")]
```

Comment:
perhaps SampleCode should be camel cased?

Reply:
Fixed in dotnet/samples#110